ntoskrnl/config/cmsysini.c

```diff
@@ -1896,11 +1899,21 @@ CmGetSystemDriverList(VOID)
         /* Get the entry */
         DriverEntry = CONTAINING_RECORD(NextEntry, BOOT_DRIVER_LIST_ENTRY, Link);
 
-        /* Allocate the path for the caller and duplicate the registry path */
+        /* Allocate the path for the caller */
         ServicePath[i] = ExAllocatePool(NonPagedPool, sizeof(UNICODE_STRING));
-        RtlDuplicateUnicodeString(RTL_DUPLICATE_UNICODE_STRING_NULL_TERMINATE,
-                                  &DriverEntry->RegistryPath,
-                                  ServicePath[i]);
+        if (!ServicePath[i])
+        {
+            KeBugCheckEx(CONFIG_INITIALIZATION_FAILED, 2, 1, 0, 0);
+        }
+
+        /* Duplicate the registry path */
+        Status = RtlDuplicateUnicodeString(RTL_DUPLICATE_UNICODE_STRING_NULL_TERMINATE,
+                                           &DriverEntry->RegistryPath,
+                                           ServicePath[i]);
+        if (!NT_SUCCESS(Status))
+        {
+            KeBugCheckEx(CONFIG_INITIALIZATION_FAILED, 2, 1, 0, 0);
```

Comment:
Doesn't make any sense compared to: https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/bug-check-0x67--config-initialization-failed
This looks like rewriting code just for rewriting code.

Reply:
I'm not sure what code you are looking at, but this change is not rewriting any code at all, it is adding a missing success check to a function that can fail and handles it with a bugcheck. The exact same bugcheck as used above, which according to your logic doesn't make any sense either.

Reply:
I'm okay with whatever bugcheck, it this feels like we should just return NULL and have the caller bugcheck (which asserts right now)?

Reply:
We could return NULL here. However, bugchecking is consistent with what is done in another part of CmGetSystemDriverList and especially the rest of the file.
If we make the KeBugCheckEx parameters unique, we could even pinpoint exactly where it failed. Way harder when just an ASSERT in the caller fails.